File: VB/EntityFrameworkServerModeExample/Form1.vb
Imports DevExpress.Data.Linq
Imports DevExpress.XtraEditors
Imports System

Namespace EntityFrameworkServerModeExample
	Partial Public Class Form1
		Inherits XtraForm

		Private entityServerModeSource As EntityServerModeSource
		Private serverMode As Boolean = False
		Public Sub New()
			InitializeComponent()
			entityServerModeSource = New EntityServerModeSource With {.ElementType = GetType(EntityInvoice), .KeyExpression = "OrderId"}
			Dim context As New NWEntities()
            context.Database.Log = AddressOf Console.Write
            entityServerModeSource.QueryableSource = context.EntityInvoices
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			' TODO: This line of code loads data into the 'nWDataSet.Invoices' table. You can move, or remove it, as needed.
			Me.invoicesTableAdapter.Fill(Me.nWDataSet.Invoices)
			SetPivotGridDataSource()
		End Sub

		Private Sub toggleSwitch1_Toggled(ByVal sender As Object, ByVal e As EventArgs) Handles toggleSwitch1.Toggled
			serverMode = DirectCast(sender, ToggleSwitch).IsOn
			SetPivotGridDataSource()
		End Sub

		Private Sub SetPivotGridDataSource()
			If serverMode Then
				pivotGridControl1.DataSource = entityServerModeSource
			Else
				pivotGridControl1.DataSource = invoicesBindingSource
			End If
		End Sub
	End Class
End Namespace

File: CS/EntityFrameworkServerModeExample/Form1.cs
using DevExpress.Data.Linq;
using DevExpress.XtraEditors;
using System;

namespace EntityFrameworkServerModeExample
{
    public partial class Form1 : XtraForm
    {
        EntityServerModeSource entityServerModeSource;
        bool serverMode = false;
        public Form1()
        {
            InitializeComponent();
            entityServerModeSource = new EntityServerModeSource
            {
                ElementType = typeof(EntityInvoice),
                KeyExpression = "OrderId"
            };
            NWEntities context = new NWEntities();
            context.Database.Log = Console.Write;
            entityServerModeSource.QueryableSource = context.EntityInvoices;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'nWDataSet.Invoices' table. You can move, or remove it, as needed.
            this.invoicesTableAdapter.Fill(this.nWDataSet.Invoices);
            SetPivotGridDataSource();
        }

        private void toggleSwitch1_Toggled(object sender, EventArgs e)
        {
            serverMode = ((ToggleSwitch)sender).IsOn;
            SetPivotGridDataSource();
        }

        private void SetPivotGridDataSource()
        {
            if (serverMode)
                pivotGridControl1.DataSource = entityServerModeSource;
            else
                pivotGridControl1.DataSource = invoicesBindingSource;
        }
    }
}
